elastic/kibana
# [Index Management] Update editable index settings for Serverless

## Summary

Fixes https://github.com/elastic/kibana/issues/165895

This PR limits which index settings are displayed on the index details page, "Settings" tab in the edit mode. On serverless only a handful of index settings will be editable by the user. The UI only prevents displaying some index settings, but it's still possible for the user to type in a setting that can't be edited. That is the case on dedicated as well. 

### How to test
1. Start Serverless ES and Kibana
2. Navigate to Index Management and create a test index
3. Click on the index name and on the details page click the tab "Settings" 
4. Toggle the "Edit mode" switch and verify that only editable settings are displayed. 

#### Screenshot
<img width="527" alt="Screenshot 2023-10-16 at 20 25 49" src="https://github.com/elastic/kibana/assets/6585477/e6678cca-3494-4c63-ae66-ace9c823d12d">
